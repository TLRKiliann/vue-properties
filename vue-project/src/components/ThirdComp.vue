<script lang="ts">
import { defineComponent } from 'vue'
export default defineComponent({
  name: "ThirdComp",
  data() {
    return {
        volume: 0,
        movie: "Superman",
        movieObj: {
            title: 'Narnia',
            movieTime: 60
        },
        movieList: ['Batman', 'SuperWoman'],
        counterTest: ''
    }
  },
  methods: {},
  computed: {},
  watch: {
    volume(newValue: number, oldValue: number): void {
        if (newValue > oldValue && newValue === 18) {
            alert("It's over 16")
        }
    },
    movie: {
        handler(newValue: string): void {
            console.log(`simple movie ${newValue}`)
        },
        immediate: true,
    },
    movieObj: {
        handler(newValue: string): void {
            console.log(`Movie object ${newValue}`)
        },
        deep: true,
    },
    movieList: {
        handler(newValue: string): void {
            console.log(`New movie ${newValue}`)
        },
        deep: true,
    }
  }
})
</script>

<template>
  <div>
    <h2>Watchers</h2>
    <h3>{{ volume }}</h3>
    <button @click="volume += 2">increment</button>
    <button @click="volume -= 2">decrement</button>
    <h3>{{ movie }}</h3>
    <input type="text" v-model="movie" />
    <h3>{{ movieObj.title }}</h3>
    <input type="text" v-model="movieObj.title" />
    <h3>{{ movieList }}</h3>
    <button @click="movieList.push('6th sens')">Add movie</button>
  </div>
</template>

<style scoped>

</style>