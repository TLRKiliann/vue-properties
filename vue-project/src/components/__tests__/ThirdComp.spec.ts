import { describe, test, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import ThirdComp from '../ThirdComp.vue'

describe('ThirdComp', () => {
  test('snapshot UI testing', () => {
    const wrapper = mount(ThirdComp,{});
    expect(wrapper.text()).toMatchSnapshot()
  })
  test('click button', async () => {
    const wrapper = mount(ThirdComp)
    await wrapper.find('button').trigger('click')
  })
})