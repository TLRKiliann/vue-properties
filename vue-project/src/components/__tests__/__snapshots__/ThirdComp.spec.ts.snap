// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ThirdComp > snapshot UI testing 1`] = `
"Watchers0incrementdecrementSupermanNarnia[
  \\"Batman\\",
  \\"SuperWoman\\"
]Add movie"
`;
